### PR TITLE
build: add support for Python 3.11

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
For the details, check the issue https://github.com/openedx/public-engineering/issues/233

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: requirements/base.txt [hunk 1]
```diff
@@ -4,15 +4,17 @@
 #
 #    make upgrade
 #
-asgiref==3.7.2
-    # via django
-backports-zoneinfo==0.2.1
+asgiref==3.8.1
     # via django
+backports-zoneinfo==0.2.1 ; python_version < "3.9"
```
Comment (felipemontoya) [inline requirements/base.txt:9]: this is the solution discussed during the maintainers call, right?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,30 +15,30 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-20.04]
-        python-version: ['3.8']
-        toxenv: ["py38", "quality", "docs"]
+        python-version: ['3.8', '3.11']
+        toxenv: ["django42", "quality", "docs"]
```
Comment (felipemontoya) [inline .github/workflows/ci.yml:19]: so this is adding support for 3.11 and also django 4.2, right?

and per the `matrix.python-version == '3.8' && matrix.toxenv == 'django42'` condition it wont fail the convo 3.8 and 4.2

File: tox.ini [hunk 3]
```diff
@@ -68,11 +69,3 @@ deps =
     -r{toxinidir}/requirements/quality.txt
 commands =
     make quality
-
```
Comment (felipemontoya) [inline tox.ini:71]: is removing the PII check part of the 3.11 upgrade?

**Review by felipemontoya — CHANGES_REQUESTED** [review]
The upgrade in itself looks good. The constraint for backports.zoneinfo looks reasonable.

This removes the PII check I left a note worth reviewing before I leave my approval.

File: tox.ini (hunk 3 above)
Comment (mariajgrimaldi) [reply tox.ini:71]: We don't use the pii checks.

File: requirements/base.txt (hunk 1 above)
Comment (mariajgrimaldi) [reply requirements/base.txt:9]: Yes: https://openedx.atlassian.net/wiki/spaces/COMM/pages/4160847886/2024-04-11+Meeting+notes

**Review by Ian2012 — APPROVED** [review]

**Review by felipemontoya — APPROVED** [review]
Now that my questions have been resolved I leave my approval as well

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-filters&entry.867055334=https://github.com/openedx/openedx-filters/pull/159&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2024-04-11+12:55&entry.1917517419=2024-04-11+22:24&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**